hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/utils/LegacyHadoopConfigurationSource.java

```diff
@@ -39,6 +40,11 @@ public String get(String key) {
     return configuration.getRaw(key);
   }
 
+  @Override
+  public char[] getPassword(String key) throws IOException {
```

Comment:
Should we also remove the deault method on ConfigurationSource for getPassword, and have the get() method in OzoneConfiguration and the new method in LegacyHadoopConfig ?

Reply:
Long-term we might need a `getPassword()` for `ConfigurationSource` and we can have different implementations (with Hadoop Configuration we can use `CredentialProviders`, with different `ConfigurationSource` we can have different approaches). 

Currently we have a very naive approach for `ConfigurationSource` we read the value as is, but using `getPassword` instead of `get` shows that we need an encryption. 

But as of know getPassword() is not called only when we use Hadoop Configuration. Agree, we can remove it temporary, until we clean-up the HTTP utilities and switch to use full `Configuration`.